homeassistant/components/automation/__init__.py

```diff
@@ -305,101 +304,68 @@ def cond_func(variables):
                     """Condition will always pass."""
                     return True
 
-            attach_triggers = partial(_process_trigger, hass, config,
-                                      config_block.get(CONF_TRIGGER, []), name)
-            entity = AutomationEntity(name, attach_triggers, cond_func, action,
-                                      hidden)
-            component.add_entities((entity,))
+            async_attach_triggers = partial(
+                _async_process_trigger, hass, config,
+                config_block.get(CONF_TRIGGER, []), name)
+            entity = AutomationEntity(name, async_attach_triggers, cond_func,
+                                      action, hidden)
+            yield from hass.loop.run_in_executor(
```

Comment:
I think we should that move out of loop. So we need not create a lot of executor and add all entities at once. The best way is to convert component_entity to async and add device with `yield from component.async_add_entities()`